src/aggregation/metric/top_hits.rs

```diff
@@ -0,0 +1,848 @@
+                    }],
+                    Default::default(),
+                ),
+            },
+        ];
+
+        let mut collector = collector_with_capacity(3);
+        for doc in docs.clone() {
+            collector.collect(doc.feature, doc.doc);
+        }
+
+        let res = collector.finalize();
+
+        assert_eq!(
+            res,
+            super::TopHitsMetricResult {
+                hits: vec![
+                    super::TopHitsVecEntry {
+                        id: docs[0].doc,
+                        sort: vec![docs[0].feature.0[0].value],
+                        search_results: Default::default(),
+                    },
+                    super::TopHitsVecEntry {
+                        id: docs[1].doc,
+                        sort: vec![docs[1].feature.0[0].value],
+                        search_results: Default::default(),
+                    },
+                    super::TopHitsVecEntry {
+                        id: docs[2].doc,
+                        sort: vec![docs[2].feature.0[0].value],
+                        search_results: Default::default(),
+                    },
+                ]
+            }
+        );
+
+        Ok(())
+    }
+
+    fn test_aggregation_top_hits(merge_segments: bool) -> crate::Result<()> {
+        let docs = vec![
+            vec![
+                r#"{ "date": "2015-01-02T00:00:00Z", "text": "bbb", "text2": "bbb" }"#,
+                r#"{ "date": "2017-06-15T00:00:00Z", "text": "ccc", "text2": "ddd" }"#,
+            ],
+            vec![
+                r#"{ "text": "aaa", "text2": "bbb", "date": "2018-01-02T00:00:00Z" }"#,
+                r#"{ "text": "aaa", "text2": "bbb", "date": "2016-01-02T00:00:00Z" }"#,
+            ],
+        ];
+
+        let index = get_test_index_from_docs(merge_segments, &docs)?;
+
+        let d: Aggregations = serde_json::from_value(json!({
+            "top_hits_req": {
+                "top_hits": {
+                    "size": 2,
+                    "sort": [
+                        { "date": "desc" }
+                    ],
+                    "from": 1,
+                    "docvalue_fields": [
+                        "date",
+                        "tex*",
+                    ],
+                }
+        }
+        }))?;
+
+        let collector = AggregationCollector::from_aggs(d, Default::default());
+        let reader = index.reader()?;
+        let searcher = reader.searcher();
+
+        let agg_res: AggregationResults = searcher.search(&AllQuery, &collector).unwrap();
+
+        let date_2017 = datetime!(2017-06-15 00:00:00 UTC);
+        let date_2016 = datetime!(2016-01-02 00:00:00 UTC);
+
+        assert_eq!(
+            agg_res
+                .0
+                .get("top_hits_req")
+                .expect("top_hits_req must exist"),
+            &AggregationResult::MetricResult(MetricResult::TopHits(TopHitsMetricResult {
+                hits: vec![
+                    TopHitsVecEntry {
```

Comment:
This is hard to read. Can you assert on the json format instead?

similar to:
```rust
        let res = exec_request_with_query(agg_req, &index, None)?;

        assert_eq!(
            res["my_stats"],
            json!({
                "avg":  2.5,
                "count": 4,
                "max": 10.0,
                "min": 0.0,
                "sum": 10.0
            })
        );

```

Reply:
done, much better!